frameworks/helloworld/tests/test_sidecar.py

```diff
@@ -54,34 +54,30 @@ def test_sidecar_parameterized():
     run_plan('sidecar-parameterized', {'PLAN_PARAMETER': 'parameterized'})
 
 
-class ToxicSidecarCheck:
+@retrying.retry(
+    wait_fixed=10000,
+    stop_max_delay=600000)
+def wait_for_toxic_sidecar_check():
     """
     Since the sidecar task fails too quickly, we check for the contents of
     the file generated in hello-container-path/toxic-output instead
 
     Note that we only check the output of hello-0.
-    """
-    @staticmethod
-    def get_cmd_output_pair():
-        """
-        In DC/OS prior to version 1.10, task exec does not run the command in the MESOS_SANDBOX directory and this
-        causes the check of the file contents to fail. Here we simply rely on the existence of the file.
-        """
-        if sdk_utils.dcos_version_less_than("1.10"):
-            cmd = "task ls hello-0-server hello-container-path/toxic-output"
-            output = ""
-        else:
-            cmd = "task exec hello-0-server cat hello-container-path/toxic-output"
-            output = "I'm addicted to you / Don't you know that you're toxic?"
 
-        return cmd, output
+    In DC/OS prior to version 1.10, task exec does not run the command in the MESOS_SANDBOX directory and this
+        causes the check of the file contents to fail. Here we simply rely on the existence of the file.
+    """
+    if sdk_utils.dcos_version_less_than("1.10"):
+        cmd = "task ls hello-0-server hello-container-path/toxic-output"
+        expected_output = ""
+    else:
+        cmd = "task exec hello-0-server cat hello-container-path/toxic-output"
+        expected_output = "I'm addicted to you / Don't you know that you're toxic?"
 
-    def __call__(self):
-        cmd, expected_output = self.get_cmd_output_pair()
-        output = sdk_cmd.run_cli(cmd).strip()
-        logging.info("Checking for toxic output returned: %s", output)
+    output = sdk_cmd.run_cli(cmd).strip()
+    logging.info("Checking for toxic output returned: %s", output)
 
-        return output == expected_output
+    assert output == expected_output
```

Comment:
There is no return statement. Does this not mean continued retries?

Reply:
all return statements are good in retrying.retry unless specific with a `retry_on_result` kwargs. assertions are the idiomatic way to trigger a retry in tests